File: 0024-Swap Nodes in Pairs/0024-Swap Nodes in Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode dummy(-1);
        ListNode* p = &dummy;
        while (head && head->next) {
            ListNode* next = head->next->next;
            p->next = head->next;
            head->next->next = head;
            head->next = next;
            
            head = next;
            p = p->next->next;
        }
        
        p->next = head;
        return dummy.next;
    }
};

File: 0001-0100/0024-Swap Nodes in Pairs/0024-Swap Nodes in Pairs.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        dummy = curr = ListNode(-1)
        while head and head.next:
            next = head.next.next
            curr.next = head.next
            head.next.next = head
            head.next = next
            
            curr = curr.next.next
            head = next
        curr.next = head
        return dummy.next
